apps/ejabberd/src/ejabberd_auth.erl

```diff
@@ -62,11 +62,7 @@
 
 -export_type([authmodule/0]).
 
--type authmodule() :: ejabberd_auth_anonymous
-                    | ejabberd_auth_external
-                    | ejabberd_auth_internal
-                    | ejabberd_auth_ldap
-                    | ejabberd_auth_odbc.
+-type authmodule() :: atom().
```

Comment:
Similar answer - I believe that this module should not care about the list of available backends. It's generic API for auth. `ejabberd_auth_http` was missing, Riak one was missing too. More may come in the future and the truth is - Dialyzer can't really verify it anyway since it comes from the config.